packages/jupyter-ai/jupyter_ai/chat_handlers/export.py

```diff
@@ -0,0 +1,33 @@
+from typing import List
+
+from jupyter_ai.models import AgentChatMessage, HumanChatMessage
+
+from .base import BaseChatHandler, SlashCommandRoutingType
+
+
+class ExportChatHandler(BaseChatHandler):
+    id = "export"
+    name = "Export chat messages"
+    help = "Export the chat messages in various formats"
+    routing_type = SlashCommandRoutingType(slash_id="export")
+
+    uses_llm = False
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+
+    def chat_message_to_markdown(self, message):
+        if isinstance(message, AgentChatMessage):
+            return f"**Agent**: {message.body}"
+        elif isinstance(message, HumanChatMessage):
+            return f"**{message.client.display_name}**: {message.body}"
+        else:
+            return ""
+
+    async def process_message(self, _):
+        markdown_content = "\n\n".join(
+            self.chat_message_to_markdown(msg) for msg in self._chat_history
+        )
+        # Write the markdown content to a file or do whatever you want with it
+        with open("./playground/chat_history.md", "w") as chat_history:
```

Comment:
This chat handler needs to handle successive `/export` calls, and the output file should not be nested under a `playground/` directory. 

I think for now, it should write to files in the following sequence when successive calls are made:

- `./chat_history.md`
- `./chat_history_1.md`
- `./chat_history_2.md`
- etc.

Reply:
I have prepared all the changes and will update this PR. All changes made to `/<jupyter-ai-top>/packages/jupyter-ai/jupyter_ai/chat_handlers/export.py`

A summary of the changes is as follows: 

1. the message says “Export the chat messages in various formats” — changed this to “Export the chat messages in markdown format”. Markdown can handle text, code, and math. Offering other formats may not in fact be ideal. Changed the PR code for this. 
![image](https://github.com/jupyterlab/jupyter-ai/assets/29005/00562136-1907-49a9-adcb-a7c87991d503)

2. I tested the PR and it works well. The only missing item is that after exporting the chat, the user is not told where the markdown file is saved, so I added code to respond with this information in the chat as shown here: 
![image](https://github.com/jupyterlab/jupyter-ai/assets/29005/a710dbc7-12da-4924-8602-2ae9ba38fc43)

3. Also updated the code to print multiple files if the same session has many /export  commands to the top level directory (not playground). It works as shown below. 
![image](https://github.com/jupyterlab/jupyter-ai/assets/29005/6b49efd3-dc61-4c6d-ad34-d61b5f0872f4)
Check that the files are at the top level directory:
![image](https://github.com/jupyterlab/jupyter-ai/assets/29005/e7ac46ab-b26f-4fd2-8018-4d656571f07b)